scp/scpHandlingEventThatInitiatesAgentSCPProgram.cpp

```diff
@@ -50,21 +63,28 @@ ScResult ASCPHandlingEventThatInitiatesAgentSCPProgram::DoProgram(ScElementaryEv
     else
     {
       SC_AGENT_LOG_WARNING(
-          "Max customer waiting time"
+          "Max customer waiting time "
           << maxCustomerWaitingTime
-          << " has expired before action of class `action_scp_interpretation_request` was initiated because"
+          << " has expired before action of class `action_scp_interpretation_request` was initiated because "
           << timeFromStart << " ms have passed");
       scpAction.InitiateAndWait();
     }
   }
-  auto const & resIt = m_context.CreateIterator5(
-      action, ScType::ConstCommonArc, ScType::ConstNode, ScType::ConstPermPosArc, ScKeynodes::nrel_result);
-  if (resIt->Next())
+
+  sc_result resultCode = SC_RESULT_OK;
+  if (action.IsFinishedSuccessfully())
+    resultCode = SC_RESULT_OK;
+  else if (action.IsFinishedUnsuccessfully())
+    resultCode = SC_RESULT_NO;
+  else if (action.IsFinishedWithError())
+    resultCode = SC_RESULT_ERROR;
+  else
   {
-    ScAddr const & actionResult = resIt->Get(2);
-    action.SetResult(actionResult);
+    action.FinishSuccessfully();
+    resultCode = SC_RESULT_OK;
   }
```

Comment:
there is one more case left
if action is finished with some status but isn't added to action_finished

Reply:
there is case when action is finished but it is not finished with some status

Reply:
I mean if action is finished successfully then this metaagent won't add it to action_finished, but all event subscriptions are subscribed to generation arc from action_finished